proto/equal.go

```diff
@@ -202,6 +202,17 @@ func equalAny(v1, v2 reflect.Value, prop *Properties) bool {
 			if v1.IsNil() != v2.IsNil() {
 				return false
 			}
+
+			// Edge case: if field is a cast type of []byte, special conversion
+			// to []byte is required for bytes.Equal.
+			if v1.Type() != reflect.TypeOf([]byte{}) {
```

Comment:
We can use the prop here to check if we are handling a casttype:
```len(prop.CastType) > 0```
And then since we know we are handling bytes. Would:
v1.Bytes()
work?